src/coreclr/src/jit/optcse.cpp

```diff
@@ -2638,17 +2638,15 @@ class CSE_Heuristic
             // If all occurances were in GT_IND nodes it could still be NO_CLASS_HANDLE
             //
             CORINFO_CLASS_HANDLE structHnd = successfulCandidate->CseDsc()->csdStructHnd;
-            assert((structHnd != NO_CLASS_HANDLE) || (cseLclVarTyp != TYP_STRUCT));
-            if (structHnd != NO_CLASS_HANDLE)
-            {
-                m_pCompiler->lvaSetStruct(cseLclVarNum, structHnd, false);
-            }
-#ifdef FEATURE_SIMD
-            else if (varTypeIsSIMD(cseLclVarTyp))
+            if (structHnd == NO_CLASS_HANDLE)
             {
-                m_pCompiler->lvaGetDesc(cseLclVarNum)->lvSIMDType = true;
```

Comment:
Do we still set lvSIMDType to true for the new CSE SIMD LclVars?
Is it done by lvaSetStruct?

Reply:
Yes,`lvaSetStruct` handles it:
https://github.com/dotnet/runtime/blob/d69cfa0f1ce8da1c6fc8ff0c2a2e088cb27a7944/src/coreclr/src/jit/lclvars.cpp#L2633-L2637